### PR TITLE
add BlockGrad op to ngraph emitter

### PR DESCRIPTION
**Pull request by ashokei**
## Description ##
add BlockGrad op to ngraph emitter

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/ngraph/ngraph_emitter.cc [hunk 1]
```diff
@@ -1508,6 +1511,10 @@ void Emitter::CreateLossOps() {
 
     return grad;
   };
+  loss_op_backward_funcs_["stop_gradient"] = [this](
+      const NodePtr& node, const NgraphNodePtr& adjoint) {
+    return makeConstant(node->inputs_[0], "0");
+  };
```
Comment (mbrookhart) [inline src/ngraph/ngraph_emitter.cc]: I don't think this will work, the manual backward function is only applied to the last node in a subgraph.

Comment (ashokei) [reply src/ngraph/ngraph_emitter.cc]: right,  i notice it fails below. we expect 0 for first output of grad. We could insert zero grad node for certain ops? Is there an optimal way to do this.

```
import mxnet as mx
v1 = mx.nd.array([[1, 2]])
v2 = mx.nd.array([[0, 1]])
a = mx.sym.Variable('a')
b = mx.sym.Variable('b')
b_stop_grad = mx.sym.stop_gradient(3 * b)
loss = mx.sym.MakeLoss(b_stop_grad + a)

executor = loss.simple_bind(ctx=mx.cpu(), a=(1,2), b=(1,2))
executor.forward(is_train=True, a=v1, b=v2)
print(executor.outputs)
executor.backward()
print(executor.grad_arrays)
```

Comment (ashokei) [reply src/ngraph/ngraph_emitter.cc]: I recall running unit-tests and this passing. I guess we cant rely on it fully.

Comment (mbrookhart) [reply src/ngraph/ngraph_emitter.cc]: Yeah, if stop_gradient is the last node in the graph, this is fine, but otherwise this fails. I think we need a stop_gradient op in nGraph, it's a no-op forward and doesn't have a defined generate adjoints.

**Comment by ashokei** [comment]
added StopGradient OP into ngraph master. Updated this PR to use that op.

**Comment by mbrookhart** [comment]
TESTNOW
